cub/cub/agent/agent_adjacent_difference.cuh

```diff
@@ -36,6 +36,9 @@
 
 #include <thrust/system/cuda/detail/core/util.h>
 
+#ifdef CCCL_AS_SYSTEM_HEADER
+_CCCL_IMPLICIT_SYSTEM_HEADER
+#endif // CCCL_AS_SYSTEM_HEADER
```

Comment:
suggest: It seems like it'd be more convenient to put the `ifdef` guard in the definition of the macro itself so this can just be a one-liner in each header as opposed to 3.

Reply:
So the issue is that cub and thrust CI fails if we do not treat libcu++ as a system header.

I believe that it would work if we move to opt out though

Reply:
Adopted to use a single line with opt out